[16.0][IMP] shopfloor: Zone picking no destination package required

lmignon opened the pull request:
Add an option to skip the packing step when processing a line into the zone picking scenario. Thanks to this option the processing of a line can be done by processing the location, set the qty and scan the destination even if the product is not into a package.

* [x] includes #900

lmignon committed:
[IMP] shopfloor: Zone picking no packing

Add an option to skip the packing step when processing a line into the zone picking scenario. Thanks to this option the processing of a line can be done by processing the location, set the qty and scan the destination even if the oroduct is not into a package.

jbaudoux commented on shopfloor/data/shopfloor_scenario_data.xml:
```diff
@@ -20,7 +20,9 @@
-    "scan_location_or_pack_first": true
+    "scan_location_or_pack_first": true,
+    "allow_alternative_destination_package": true,
+    "no_packing": true
```
I don't like negative options. And as we already have `destination_package` (in fact there is already 2 names for the same thing: pack & package), I would prefer
```suggestion
    "require_destination_package": true
```

lmignon commented:
@jbaudoux Code changed

sbejaoui committed:
[IMP] shopfloor, zone picking: Allow partial transfer when scanning a location.

In the zone picking scenario, operators can choose to pick less than the reserved quantity.
Previously, the system required them, in this case, to place the goods in a package first.

This restriction was unnecessary, as it forced operators to perform an extra step that they do not actually do in real life.
We trust them to know what they're doing, so let's keep it simple.

With this fix, if a user scans a location, they can move the products directly to the scanned location.

rousseldenis commented:
@jbaudoux

lmignon commented:
@jbaudoux rebased to fix conflicts. Could be merged....

jbaudoux commented:
@lmignon there is a small pre-commit issue

lmignon commented:
@jbaudoux pre-commit is now :green_circle: :smirk:

jbaudoux commented:
cc @sebalix @simahawk @TDu

rousseldenis commented:
/ocabot merge nobump
